crowd_sim/envs/crowd_sim.py

```diff
@@ -365,10 +489,20 @@ def step(self, action, update=True):
         end_position = np.array(self.robot.compute_position(action, self.time_step))
         reaching_goal = norm(end_position - np.array(self.robot.get_goal_position())) < self.robot.radius
 
+        ## check if the robot run out of the boundary
+        robot_x, robot_y = self.robot.get_position()
+        out = False
+        if np.abs(robot_x) > self.boundary / 2 or np.abs(robot_y) > self.boundary / 2:
+            out = True
+
         if self.global_time >= self.time_limit - 1:
             reward = 0
             done = True
             info = Timeout()
+        elif out:
+            reward = self.out_boundary_penalty
```

Comment:
For future implementation:
It`s ok to merge it like this, but after we add rectangular obstacles: instead of out of boundary penalty, can we add the boundary as an obstacle?

Reply:
Good suggestion. Today, when I was in class I keep thinking about the rectangle obstacles' implementation. Because now either the obstacles or humans follow ORCA so that they don't crash into others, but once we implement the obstacles with a rectangle shape, I think there would be a bunch of things that need to reconsider lol. I haven't started it yet. But modeling the boundary as an obstacle is actually a good idea! I would keep this in mind.

Reply:
Oh right.. the changes to orca was completely out of consideration. We might need to derive a distance metric that takes 4 edges of a rectangle as input. Something like this:
```
def dist_to_rectangle(rectangle,agent):
    lines.append(ine_between(rectangle.edges[0],rectangle.edges[1]))
    lines.append(line_between(rectangle.edges[1],rectangle.edges[2]))
    ...
    min_dist = inf
    for line in lines:
        dist = dist_to_line(agent,line)
        if dist < min_dist:
             min_dist = dist
    return min_dist
```

Reply:
We discuss further on this during the next merge, resolving this for now